idaes/surrogate/alamopy_new.py

```diff
@@ -0,0 +1,673 @@
+        ))
+    CONFIG.declare('tolabsmetric', ConfigValue(
+        default=None,
+        domain=list_of_floats,
+        description="Absolute tolerance for outputs.",
+        doc="Absolute convergence tolerance for the chosen fitness metric for "
+        "the modeling of output variables. One per output variable, space "
+        "separated. Incremental model building will stop if two consecutive "
+        "iterations do not improve the chosen metric by at least this amount."
+        ))
+    CONFIG.declare('tolmeanerror', ConfigValue(
+        default=None,
+        domain=list_of_floats,
+        description="Convergence tolerance for mean errors in outputs.",
+        doc="Row vector of convergence tolerances for mean errors in the "
+        "modeling of output variables. One per output variable, space "
+        "separated. Incremental model building will stop if tolmeanerror, "
+        "tolrelmetric, or tolabsmetric is satisfied."))
+    CONFIG.declare('tolsse', ConfigValue(
+        default=None,
+        domain=float,
+        description="Absolute tolerance on SSE",
+        doc="Absolute tolerance on sum of square errors (SSE). ALAMO will "
+        "terminate if it finds a solution whose SSE is within tolsse from "
+        "the SSE of the full least squares problem."))
+
+    CONFIG.declare('mipoptca', ConfigValue(
+        default=None,
+        domain=float,
+        description="Absolute tolerance for MIP."))
+    CONFIG.declare('mipoptcr', ConfigValue(
+        default=None,
+        domain=float,
+        description="Relative tolerance for MIP."))
+    CONFIG.declare('linearerror', ConfigValue(
+        default=None,
+        domain=In([True, False]),
+        description="If True, a linear objective is used when solving "
+        "mixed-integer optimization problems; otherwise, a squared error will "
+        "be employed."))
+    CONFIG.declare('GAMS', ConfigValue(
+        default=None,
+        domain=str,
+        description="Complete path of GAMS executable (or name if GAMS is in "
+        "the user path)."))
+    CONFIG.declare('GAMSSOLVER', ConfigValue(
+        default=None,
+        domain=str,
+        description="Name of preferred GAMS solver for solving ALAMO’s "
+        "mixed-integer quadratic subproblems. Special facilities have been "
+        "implemented in ALAMO and BARON that make BARON the preferred "
+        "selection for this option. However, any mixed-integer quadratic "
+        "programming solver available under GAMS can be used."))
+    CONFIG.declare('solvemip', ConfigValue(
+        default=None,
+        domain=In([True, False]),
+        description="Whether to use an optimizer to solve MIP."))
+    # CONFIG.declare('log_output', ConfigValue(
+    #     default=False,
+    #     domain=In([True, False]),
+    #     description="Whether to log ALAMO output."))
+
+    # I/O file options
+    CONFIG.declare("temp_path", ConfigValue(
+        default=DEFAULTPATH,
+        domain=str,

```

Comment:
If you are going to do this here, I recommend the `Path` domain and not just `str`)